cmake/HalideGeneratorHelpers.cmake

```diff
@@ -515,8 +514,28 @@ endfunction()
 # Function for creating a standalone runtime from a generator.
 ##
 
-function(_Halide_add_halide_runtime RT)
-    cmake_parse_arguments(ARG "" "FROM" "TARGETS" ${ARGN})
+function(add_halide_runtime RT)
+    set(options "")
+    set(oneValueArgs "")
+    set(multiValueArgs TARGETS)
+    cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+
+    # If no TARGETS argument, use Halide_TARGET instead
+    if (NOT ARG_TARGETS)
+        set(ARG_TARGETS "${Halide_TARGET}")
+    endif ()
+
+    # Ensure all targets are tagged with "no_runtime",
+    # so that GCD calculation doesn't get confused.
+    list(TRANSFORM ARG_TARGETS APPEND "-no_runtime")
+
+    # Create a Generator that is GenGen.cpp and nothing else; all it can do is generate a runtime.
```

Comment:
TODO: ship this and don't build it lazily like this

Reply:
...I don't understand?